tests/Authenticator.hs

```diff
@@ -49,17 +55,52 @@ data AuthenticatorSignatureCounter
   = Unsupported
   | Global M.SignatureCounter
   | PerCredential (Map.Map M.CredentialId M.SignatureCounter)
-  deriving (Show)
+
+data SigningBehaviour
+  = GenerateCorrectSignature
+  | GenerateRandomSignature
+
+data RpIdHashBehaviour
+  = GenerateCorrectHash
+  | GenerateRandomHash
```

Comment:
Same here, and I think because of that this distinction (and this enum) isn't really needed. Instead it should be a distinction of the `RpId` that is being hashed. So something like
```haskell
data RpIdBehaviour
  = RpIdBehaviourCorrect
  | RpIdBehaviourWrong
```

In case of `Wrong`, it should perhaps use a random RpId